src/libhirte/test/common/network/get_address_test.c

```diff
@@ -7,10 +7,45 @@
 #include "libhirte/common/network.h"
 #include "libhirte/common/string-util.h"
 
+int getaddrinfo_func_mock_failure(const char *, const char *, const struct addrinfo *, struct addrinfo **) {
+        return -1;
+}
+
+int getaddrinfo_func_mock(const char *, const char *, const struct addrinfo *, struct addrinfo **foo) {
```

Comment:
Just wondering if it wouldn't be better to have more complete mock definition?

1. Have a mocking parameters and results structure:
````
struct getaddrinfo_mock_params {
    char* hostname;
    char** resolved_addresses;
    int result;
}
````
2. Initialize this structure for all test cases:
````
static struct getaddrinfo_mock_params[] test_data = [
    { NULL, NULL, -1 },
    { ".", NULL, -1 },
    { "redhat", NULL, -1 },
    { "?10.10.10.3", NULL -1 },
    { "192.168.1.", NULL, -1 },
    { "8.8.8.8", ["8.8.8.8"], 0},
    { "localhost.localdomain", ["127.0.01", "::1"], 0}
];
````
3. Having only `getaddrinfo_func_mock`, which uses `test_data` array and create output according to the input and `test_data` content